### ISSUE TITLE
Name 'id' must match pattern '^[a-z]{2,}[a-zA-Z]+$'. (MethodNameCheck)

### ISSUE THREAD

**Issue by dgroup** [teamed/qulice#960]
Hi Guys,
What is the business purpose of this check for the fields and methods with name equal to `id`?
This name quite self-explaining.

The same is related to PMD
- `Field id has the same name as a method (AvoidFieldNameMatchingMethodName)`
- `Avoid using short method names (ShortMethodName)`

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) please, pay attention to this issue

**Comment by 0crat** [comment]
@dgroup[/z](https://www.0crat.com/u/dgroup) this project will fix the problem faster if you donate a few dollars to it; just [click here](https://www.0crat.com/contrib/C3T49A35L) and pay via Stripe, it's very fast, convenient and appreciated; thanks a lot!

**Comment by krzyk** [comment]
@dgroup makes sense (considering that we already have https://github.com/teamed/qulice/issues/960)

**Comment by krzyk** [comment]
@0crat in

**Comment by dgroup** [comment]
@krzyk no response from 0crat more than 7 min. Looks like 0crat went into the infinitive loop due to your `#960` self-link :)

**Comment by 0crat** [comment]
> @0crat in ([here](https://github.com/teamed/qulice/issues/960#issuecomment-445178970))

@krzyk Job [#960](https://github.com/teamed/qulice/issues/960) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
The job [#960](https://github.com/teamed/qulice/issues/960) assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/C3T49A35L/dd4c660e-362a-493a-bb9d-59591c9f6bc8); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by paulodamaso** [comment]
@krzyk I believe that it has already been resolved in the linked PRs, what should I do here?

**Comment by krzyk** [comment]
@paulodamaso there wasn't any PR here, the linked issue is for slightly different problem

**Comment by paulodamaso** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/teamed/qulice/issues/960#issuecomment-460069008))

@paulodamaso The user @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) resigned from [#960](https://github.com/teamed/qulice/issues/960), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#960](https://github.com/teamed/qulice/issues/960); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#960](https://github.com/teamed/qulice/issues/960); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#960](https://github.com/teamed/qulice/issues/960); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#960](https://github.com/teamed/qulice/issues/960); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#960](https://github.com/teamed/qulice/issues/960); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#960](https://github.com/teamed/qulice/issues/960); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#960](https://github.com/teamed/qulice/issues/960); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#960](https://github.com/teamed/qulice/issues/960); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#960](https://github.com/teamed/qulice/issues/960); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by yegor256** [comment]
Fixed in 038b18b39.

Made three changes:

1. **Checkstyle `MethodName`** (`src/main/resources/com/qulice/checkstyle/checks.xml`) — widened the regex from `^[a-z]{2,}[a-zA-Z]+$` to `^(id|[a-z]{2,}[a-zA-Z]+)$`, matching the pattern already used for `MemberName`, `LocalVariableName`, `LocalFinalVariableName` and `ParameterName`.

2. **PMD `ShortMethodName`** (`src/main/resources/com/qulice/pmd/ruleset.xml`) — excluded, alongside the already-excluded `ShortVariable` and `ShortClassName`.

3. **PMD `AvoidFieldNameMatchingMethodName`** (`src/main/resources/com/qulice/pmd/ruleset.xml`) — excluded, since the idiomatic EO pattern `private final T id; public T id()` would otherwise always fire it.

Added one checkstyle test (`CheckstyleValidatorTest#acceptsIdAsMethodName`) and two PMD disabled-rule tests, with fixtures for each.

Note: `name` (4 chars) already passes all checks — only `id` was rejected.
